### PR TITLE
certbot: 1.3.0 -> 1.6.0

### PR DESCRIPTION
**Pull request by flokli**
This includes https://github.com/certbot/certbot/pull/8081, so we can
stop shipping our custom patch to workaroung things, and start packaging
more certbot plugins.

Other release notes:
 - https://github.com/certbot/certbot/releases/tag/v1.4.0
 - https://github.com/certbot/certbot/releases/tag/v1.5.0
 - https://github.com/certbot/certbot/releases/tag/v1.6.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by flokli** [comment]
I added 3 common DNS plugins I might end up using - the others should be simple enough to be packaged once someone starts using them.

I was able to access these plugins from a `nix-shell` containing the additional python package, so I assume adding them to `PYTHONPATH` is all that's needed.

@FRidh, @domenkozar, do you have any opinions on a `certbotWithPackages`, `certbot-full` or similar?

**Comment by flokli** [comment]
cc @NixOS/acme

**Comment by m1cr0man** [comment]
Is the intention long-term to replace lego with cerbot? If so, I might just do it during my rewrite.

**Comment by flokli** [comment]
This is unrelated to the acme module system - I have some different tooling currently hard-wired to certbot and using some of the plugins packaged here - I still thought it might make sense to highlight the acme team.

**Comment by NinjaTrappeur** [comment]
`nix-build -A pythonPackages.certbot-dns-cloudflare` is failing.

We probably remove these packages from the python2 packageset.

File: pkgs/development/python-modules/certbot-dns-cloudflare/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  ];
+
+  checkInputs = [
+    pytest
+    pytestCheckHook
+  ];
+
+  disabled = !isPy3k;
+
+  pytestFlagsArray = [ "-o cache_dir=/tmp" ];
```
Comment (NinjaTrappeur) [inline pkgs/development/python-modules/certbot-dns-cloudflare/default.nix:28]:
```suggestion
  pytestFlagsArray = [ "-o cache_dir=$(mktemp -d)" ];
```

Will probably make this slightly more portable.

File: pkgs/development/python-modules/certbot-dns-rfc2136/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  ];
+
+  checkInputs = [
+    pytest
+    pytestCheckHook
+  ];
+
+  disabled = !isPy3k;
+
+  pytestFlagsArray = [ "-o cache_dir=/tmp" ];
```
Comment (NinjaTrappeur) [inline pkgs/development/python-modules/certbot-dns-rfc2136/default.nix:28]:
```suggestion
  pytestFlagsArray = [ "-o cache_dir=$(mktemp -d)" ];
```

Will probably make this slightly more portable.

File: pkgs/development/python-modules/certbot-dns-route53/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  ];
+
+  checkInputs = [
+    pytest
+    pytestCheckHook
+  ];
+
+  disabled = !isPy3k;
+
+  pytestFlagsArray = [ "-o cache_dir=/tmp" ];
```
Comment (NinjaTrappeur) [inline pkgs/development/python-modules/certbot-dns-route53/default.nix:28]:
```suggestion
  pytestFlagsArray = [ "-o cache_dir=$(mktemp -d)" ];
```

Will probably make this slightly more portable.

**Review by picnoir — COMMENTED** [review]
Portability nitpick.

Other than that, LGTM

File: pkgs/development/python-modules/certbot-dns-cloudflare/default.nix (hunk 1 above)
Comment (flokli) [reply pkgs/development/python-modules/certbot-dns-cloudflare/default.nix]: fixed.

File: pkgs/development/python-modules/certbot-dns-route53/default.nix (hunk 3 above)
Comment (flokli) [reply pkgs/development/python-modules/certbot-dns-route53/default.nix]: fixed.

File: pkgs/development/python-modules/certbot-dns-rfc2136/default.nix (hunk 2 above)
Comment (flokli) [reply pkgs/development/python-modules/certbot-dns-rfc2136/default.nix]: fixed.

**Review by picnoir — APPROVED** [review]
LGTM

**Comment by flokli** [comment]
@GrahamcOfBorg build certbot

**Pull request merged**